lib/src/test/kotlin/org/didcommx/didcomm/EncryptedMessageTest.kt

```diff
@@ -128,4 +136,71 @@ class EncryptedMessageTest {
             assertEquals(tv.expectedMessage, actual.message)
         }
     }
+
+    @Test
+    fun `Test_encrypt_decrypt_message_with_from_prior_and_issuer_kid`() {
+        val didComm = DIDComm(DIDDocResolverMock(), AliceRotatedToCharlieSecretResolverMock())
+
+        for (message in listOf(JWM.PLAINTEXT_MESSAGE_FROM_PRIOR_MINIMAL, JWM.PLAINTEXT_MESSAGE_FROM_PRIOR)) {
+            val packResult = didComm.packEncrypted(
+                PackEncryptedParams.builder(message, JWM.BOB_DID)
+                    .from(JWM.CHARLIE_DID)
+                    .fromPriorIssuerKid("did:example:alice#key-2")
+                    .build()
+            )
+
+            assertNotNull(packResult.packedMessage)
+            assertEquals("did:example:alice#key-2", packResult.fromPriorIssuerKid)
+
+            val unpackResult = didComm.unpack(
+                UnpackParams.Builder(packResult.packedMessage)
+                    .secretResolver(BobSecretResolverMock())
+                    .build()
+            )
+
+            assertEquals(message, unpackResult.message)
+            assertEquals("did:example:alice#key-2", unpackResult.metadata.fromPriorIssuerKid)
+            assertNotNull(unpackResult.metadata.fromPriorJwt)
+        }
+    }
+
+    @Test
+    fun `Test_encrypt_decrypt_message_with_from_prior_and_no_issuer_kid`() {
+        val didComm = DIDComm(DIDDocResolverMock(), AliceRotatedToCharlieSecretResolverMock())
+
+        for (message in listOf(JWM.PLAINTEXT_MESSAGE_FROM_PRIOR_MINIMAL, JWM.PLAINTEXT_MESSAGE_FROM_PRIOR)) {
+            val packResult = didComm.packEncrypted(
+                PackEncryptedParams.builder(message, JWM.BOB_DID)
+                    .from(JWM.CHARLIE_DID)
+                    .build()
+            )
+
+            assertNotNull(packResult.packedMessage)
+            assertNotNull(packResult.fromPriorIssuerKid)
+            assertTrue(isDID(packResult.fromPriorIssuerKid!!))
+            assertTrue(isDIDFragment(packResult.fromPriorIssuerKid!!))
+            assertEquals(JWM.ALICE_DID, divideDIDFragment(packResult.fromPriorIssuerKid!!).first())
+
+            val unpackResult = didComm.unpack(
+                UnpackParams.Builder(packResult.packedMessage)
+                    .secretResolver(BobSecretResolverMock())
+                    .build()
+            )
+
+            assertEquals(message, unpackResult.message)
+            assertEquals(packResult.fromPriorIssuerKid, unpackResult.metadata.fromPriorIssuerKid)
+            assertNotNull(unpackResult.metadata.fromPriorJwt)
+        }
+    }
+
+    @Test
+    fun `Test_encrypt_message_with_invalid_from_prior`() {
+        val didComm = DIDComm(DIDDocResolverMock(), AliceRotatedToCharlieSecretResolverMock())
+
+        for (message in JWM.INVALID_FROM_PRIOR_PLAINTEXT_MESSAGES) {
+            assertFails {
```

Comment:
Please expect an appropriate exception class, not just fail (to make sure that we throw a proper exception, not just fail)